<?xml version="1.0" encoding="UTF-8"?>
<testsuite xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="https://maven.apache.org/surefire/maven-surefire-plugin/xsd/surefire-test-report-3.0.xsd" version="3.0" name="com.junit.exception.ExceptionThrowerTest" time="0.459" tests="24" errors="0" skipped="0" failures="0">
  <properties>
    <property name="java.specification.version" value="24"/>
    <property name="sun.cpu.isalist" value="amd64"/>
    <property name="sun.jnu.encoding" value="Cp1252"/>
    <property name="java.class.path" value="c:\Users\saika\OneDrive\Desktop\Cognizant\Digital-Nuture-4.0-JavaFSE\Week-2_TDD using JUnit5 and Mockito\2. JUnit_Advanced Testing exercices\Exercise 4 Exception Testing\ExceptionTestingExample\target\test-classes;c:\Users\saika\OneDrive\Desktop\Cognizant\Digital-Nuture-4.0-JavaFSE\Week-2_TDD using JUnit5 and Mockito\2. JUnit_Advanced Testing exercices\Exercise 4 Exception Testing\ExceptionTestingExample\target\classes;C:\Users\saika\.m2\repository\org\junit\jupiter\junit-jupiter\5.10.1\junit-jupiter-5.10.1.jar;C:\Users\saika\.m2\repository\org\junit\jupiter\junit-jupiter-api\5.10.1\junit-jupiter-api-5.10.1.jar;C:\Users\saika\.m2\repository\org\opentest4j\opentest4j\1.3.0\opentest4j-1.3.0.jar;C:\Users\saika\.m2\repository\org\junit\platform\junit-platform-commons\1.10.1\junit-platform-commons-1.10.1.jar;C:\Users\saika\.m2\repository\org\apiguardian\apiguardian-api\1.1.2\apiguardian-api-1.1.2.jar;C:\Users\saika\.m2\repository\org\junit\jupiter\junit-jupiter-params\5.10.1\junit-jupiter-params-5.10.1.jar;C:\Users\saika\.m2\repository\org\junit\jupiter\junit-jupiter-engine\5.10.1\junit-jupiter-engine-5.10.1.jar;C:\Users\saika\.m2\repository\org\junit\platform\junit-platform-engine\1.10.1\junit-platform-engine-1.10.1.jar;"/>
    <property name="java.vm.vendor" value="Eclipse Adoptium"/>
    <property name="sun.arch.data.model" value="64"/>
    <property name="user.variant" value=""/>
    <property name="java.vendor.url" value="https://adoptium.net/"/>
    <property name="user.timezone" value="Asia/Calcutta"/>
    <property name="os.name" value="Windows 11"/>
    <property name="java.vm.specification.version" value="24"/>
    <property name="sun.java.launcher" value="SUN_STANDARD"/>
    <property name="user.country" value="IN"/>
    <property name="sun.boot.library.path" value="C:\Users\saika\AppData\Roaming\Code\User\globalStorage\pleiades.java-extension-pack-jdk\java\latest\bin"/>
    <property name="sun.java.command" value="C:\Users\saika\AppData\Local\Temp\surefire92919728998531935\surefirebooter-20250625204138161_3.jar C:\Users\saika\AppData\Local\Temp\surefire92919728998531935 2025-06-25T20-41-37_741-jvmRun1 surefire-20250625204138161_1tmp surefire_0-20250625204138161_2tmp"/>
    <property name="jdk.debug" value="release"/>
    <property name="surefire.test.class.path" value="c:\Users\saika\OneDrive\Desktop\Cognizant\Digital-Nuture-4.0-JavaFSE\Week-2_TDD using JUnit5 and Mockito\2. JUnit_Advanced Testing exercices\Exercise 4 Exception Testing\ExceptionTestingExample\target\test-classes;c:\Users\saika\OneDrive\Desktop\Cognizant\Digital-Nuture-4.0-JavaFSE\Week-2_TDD using JUnit5 and Mockito\2. JUnit_Advanced Testing exercices\Exercise 4 Exception Testing\ExceptionTestingExample\target\classes;C:\Users\saika\.m2\repository\org\junit\jupiter\junit-jupiter\5.10.1\junit-jupiter-5.10.1.jar;C:\Users\saika\.m2\repository\org\junit\jupiter\junit-jupiter-api\5.10.1\junit-jupiter-api-5.10.1.jar;C:\Users\saika\.m2\repository\org\opentest4j\opentest4j\1.3.0\opentest4j-1.3.0.jar;C:\Users\saika\.m2\repository\org\junit\platform\junit-platform-commons\1.10.1\junit-platform-commons-1.10.1.jar;C:\Users\saika\.m2\repository\org\apiguardian\apiguardian-api\1.1.2\apiguardian-api-1.1.2.jar;C:\Users\saika\.m2\repository\org\junit\jupiter\junit-jupiter-params\5.10.1\junit-jupiter-params-5.10.1.jar;C:\Users\saika\.m2\repository\org\junit\jupiter\junit-jupiter-engine\5.10.1\junit-jupiter-engine-5.10.1.jar;C:\Users\saika\.m2\repository\org\junit\platform\junit-platform-engine\1.10.1\junit-platform-engine-1.10.1.jar;"/>
    <property name="sun.cpu.endian" value="little"/>
    <property name="user.home" value="C:\Users\saika"/>
    <property name="user.language" value="en"/>
    <property name="java.specification.vendor" value="Oracle Corporation"/>
    <property name="java.version.date" value="2025-04-15"/>
    <property name="java.home" value="C:\Users\saika\AppData\Roaming\Code\User\globalStorage\pleiades.java-extension-pack-jdk\java\latest"/>
    <property name="file.separator" value="\"/>
    <property name="basedir" value="c:\Users\saika\OneDrive\Desktop\Cognizant\Digital-Nuture-4.0-JavaFSE\Week-2_TDD using JUnit5 and Mockito\2. JUnit_Advanced Testing exercices\Exercise 4 Exception Testing\ExceptionTestingExample"/>
    <property name="java.vm.compressedOopsMode" value="32-bit"/>
    <property name="line.separator" value="&#10;"/>
    <property name="java.vm.specification.vendor" value="Oracle Corporation"/>
    <property name="java.specification.name" value="Java Platform API Specification"/>
    <property name="surefire.real.class.path" value="C:\Users\saika\AppData\Local\Temp\surefire92919728998531935\surefirebooter-20250625204138161_3.jar"/>
    <property name="user.script" value=""/>
    <property name="sun.management.compiler" value="HotSpot 64-Bit Tiered Compilers"/>
    <property name="java.runtime.version" value="24.0.1+9"/>
    <property name="user.name" value="saika"/>
    <property name="stdout.encoding" value="Cp1252"/>
    <property name="path.separator" value=";"/>
    <property name="os.version" value="10.0"/>
    <property name="java.runtime.name" value="OpenJDK Runtime Environment"/>
    <property name="file.encoding" value="UTF-8"/>
    <property name="java.vm.name" value="OpenJDK 64-Bit Server VM"/>
    <property name="java.vendor.version" value="Temurin-24.0.1+9"/>
    <property name="localRepository" value="C:\Users\saika\.m2\repository"/>
    <property name="java.vendor.url.bug" value="https://github.com/adoptium/adoptium-support/issues"/>
    <property name="java.io.tmpdir" value="C:\Users\saika\AppData\Local\Temp\"/>
    <property name="java.version" value="24.0.1"/>
    <property name="user.dir" value="c:\Users\saika\OneDrive\Desktop\Cognizant\Digital-Nuture-4.0-JavaFSE\Week-2_TDD using JUnit5 and Mockito\2. JUnit_Advanced Testing exercices\Exercise 4 Exception Testing\ExceptionTestingExample"/>
    <property name="os.arch" value="amd64"/>
    <property name="java.vm.specification.name" value="Java Virtual Machine Specification"/>
    <property name="sun.os.patch.level" value=""/>
    <property name="native.encoding" value="Cp1252"/>
    <property name="java.library.path" value="C:\Users\saika\AppData\Roaming\Code\User\globalStorage\pleiades.java-extension-pack-jdk\java\latest\bin;C:\WINDOWS\Sun\Java\bin;C:\WINDOWS\system32;C:\WINDOWS;C:\Users\saika\AppData\Roaming\Code\User\globalStorage\pleiades.java-extension-pack-jdk\gradle\latest\bin;C:\Users\saika\AppData\Roaming\Code\User\globalStorage\pleiades.java-extension-pack-jdk\maven\latest\bin;C:\Users\saika\AppData\Roaming\Code\User\globalStorage\pleiades.java-extension-pack-jdk\java\latest\bin;C:\hive\bin;C:\hadoop\bin;C:\hadoop\sbin;C:\Program Files\Java\bin;C:\Program Files\Eclipse Adoptium\jdk-8.0.452.9-hotspot\bin;c:\Users\saika\AppData\Local\Programs\cursor\resources\app\bin;C:\Python\Scripts\;C:\Python\;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Windows\System32\OpenSSH\;C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common;C:\Program Files\NVIDIA Corporation\NVIDIA NvDLISR;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\Program Files\Git\cmd;C:\Program Files\Git\bin;C:\Program Files\nodejs\;C:\ProgramData\chocolatey\bin;C:\derby\bin;C:\Program Files\Eclipse Adoptium\jdk-11.0.26.4-hotspot\bin;C:\Users\saika\.local\bin;C:\Users\saika\anaconda3;C:\Users\saika\anaconda3\Library\mingw-w64\bin;C:\hadoop\bin;C:\Program Files\MySQL\MySQL Server 5.7\bin;C:\hive\bin;C:\hive\lib;C:\hive\bin;C:\derby\bin;C:\Program Files\Eclipse Adoptium\jdk-8.0.452.9-hotspot\bin;C:\Users\saika\anaconda3\Library\usr\bin;C:\Users\saika\ana;C:\Windows\System32;C:\hadoop\bin;C:\Users\saika\anaconda3;C:\Users\saika\anaconda3\Library\mingw-w64\bin;C:\Users\saika\anaconda3\Library\usr\bin;C:\Users\saika\anaconda3\Library\bin;C:\Users\saika\anaconda3\Scripts;C:\Users\saika\AppData\Local\Microsoft\WindowsApps;C:\Microsoft VS Code\bin;C:\Users\saika\AppData\Roaming\npm;C:\Program Files\JetBrains\PyCharm Community Edition 2024.3.1.1\bin;;C:\Users\saika\AppData\Local\Programs\cursor\resources\app\bin;C:\Users\saika\.cargo\bin;C:\hive\bin;C:\Program Files\Java\bin;C:\Program Files\MySQL\MySQL Server 5.7\bin;;c:\Users\saika\AppData\Roaming\Code\User\globalStorage\github.copilot-chat\debugCommand;."/>
    <property name="java.vm.info" value="mixed mode, sharing"/>
    <property name="stderr.encoding" value="Cp1252"/>
    <property name="java.vendor" value="Eclipse Adoptium"/>
    <property name="java.vm.version" value="24.0.1+9"/>
    <property name="sun.io.unicode.encoding" value="UnicodeLittle"/>
    <property name="java.class.version" value="68.0"/>
  </properties>
  <testcase name="testValidateInputWithEmpty" classname="com.junit.exception.ExceptionThrowerTest" time="0.143">
    <system-out><![CDATA[Setting up test for exception testing...
Testing validateInput with empty input...
✓ IllegalArgumentException thrown for empty input
Test completed.

]]></system-out>
  </testcase>
  <testcase name="testProcessAgeNegative" classname="com.junit.exception.ExceptionThrowerTest" time="0.017">
    <system-out><![CDATA[Setting up test for exception testing...
Testing processAge with negative age...
✓ InvalidAgeException thrown for negative age
Test completed.

]]></system-out>
  </testcase>
  <testcase name="testMultipleExceptionsWithAssertAll" classname="com.junit.exception.ExceptionThrowerTest" time="0.023">
    <system-out><![CDATA[Setting up test for exception testing...
Testing multiple exceptions with assertAll...
✓ All exception assertions passed
Test completed.

]]></system-out>
  </testcase>
  <testcase name="testDivideByZero" classname="com.junit.exception.ExceptionThrowerTest" time="0.005">
    <system-out><![CDATA[Setting up test for exception testing...
Testing division by zero...
✓ ArithmeticException thrown for division by zero
Test completed.

]]></system-out>
  </testcase>
  <testcase name="testProcessAgeValid" classname="com.junit.exception.ExceptionThrowerTest" time="0.016">
    <system-out><![CDATA[Setting up test for exception testing...
Testing processAge with valid age...
✓ No exception thrown for valid age
Test completed.

]]></system-out>
  </testcase>
  <testcase name="testMultipleExceptionScenariosNull" classname="com.junit.exception.ExceptionThrowerTest" time="0.005">
    <system-out><![CDATA[Setting up test for exception testing...
Testing multiple exception scenarios - null...
✓ NullPointerException thrown for null scenario
Test completed.

]]></system-out>
  </testcase>
  <testcase name="testProcessAgeTooHigh" classname="com.junit.exception.ExceptionThrowerTest" time="0.009">
    <system-out><![CDATA[Setting up test for exception testing...
Testing processAge with age too high...
✓ InvalidAgeException thrown for age too high
Test completed.

]]></system-out>
  </testcase>
  <testcase name="testValidateInputWithValidInput" classname="com.junit.exception.ExceptionThrowerTest" time="0.004">
    <system-out><![CDATA[Setting up test for exception testing...
Testing validateInput with valid input...
✓ No exception thrown for valid input
Test completed.

]]></system-out>
  </testcase>
  <testcase name="testParseNumberValid" classname="com.junit.exception.ExceptionThrowerTest" time="0.016">
    <system-out><![CDATA[Setting up test for exception testing...
Testing parseNumber with valid input...
✓ Valid number parsed: 123
Test completed.

]]></system-out>
  </testcase>
  <testcase name="testProcessDataThrowException" classname="com.junit.exception.ExceptionThrowerTest" time="0.01">
    <system-out><![CDATA[Setting up test for exception testing...
Testing processData that should throw exception...
✓ UnsupportedOperationException thrown as expected
Test completed.

]]></system-out>
  </testcase>
  <testcase name="testValidateInputWithNull" classname="com.junit.exception.ExceptionThrowerTest" time="0.009">
    <system-out><![CDATA[Setting up test for exception testing...
Testing validateInput with null input...
✓ IllegalArgumentException thrown for null input
Test completed.

]]></system-out>
  </testcase>
  <testcase name="testGetArrayElementNullArray" classname="com.junit.exception.ExceptionThrowerTest" time="0.005">
    <system-out><![CDATA[Setting up test for exception testing...
Testing getArrayElement with null array...
✓ NullPointerException thrown for null array
Test completed.

]]></system-out>
  </testcase>
  <testcase name="testMultipleExceptionScenariosNoException" classname="com.junit.exception.ExceptionThrowerTest" time="0.004">
    <system-out><![CDATA[Setting up test for exception testing...
Testing multiple exception scenarios - normal...
✓ No exception thrown for normal scenario
Test completed.

]]></system-out>
  </testcase>
  <testcase name="testMultipleExceptionScenariosIllegal" classname="com.junit.exception.ExceptionThrowerTest" time="0.004">
    <system-out><![CDATA[Setting up test for exception testing...
Testing multiple exception scenarios - illegal...
✓ IllegalArgumentException thrown for illegal scenario
Test completed.

]]></system-out>
  </testcase>
  <testcase name="testParseNumberWithNull" classname="com.junit.exception.ExceptionThrowerTest" time="0.005">
    <system-out><![CDATA[Setting up test for exception testing...
Testing parseNumber with null input...
✓ NumberFormatException thrown for null input
Test completed.

]]></system-out>
  </testcase>
  <testcase name="testProcessDataNoException" classname="com.junit.exception.ExceptionThrowerTest" time="0.005">
    <system-out><![CDATA[Setting up test for exception testing...
Testing processData that should not throw exception...
✓ No exception thrown, result: TEST
Test completed.

]]></system-out>
  </testcase>
  <testcase name="testGetArrayElementInvalidIndex" classname="com.junit.exception.ExceptionThrowerTest" time="0.008">
    <system-out><![CDATA[Setting up test for exception testing...
Testing getArrayElement with invalid index...
✓ IndexOutOfBoundsException thrown for invalid index
Test completed.

]]></system-out>
  </testcase>
  <testcase name="testExceptionHierarchy" classname="com.junit.exception.ExceptionThrowerTest" time="0.004">
    <system-out><![CDATA[Setting up test for exception testing...
Testing exception hierarchy...
✓ Exception hierarchy test passed
Test completed.

]]></system-out>
  </testcase>
  <testcase name="testCheckStateValid" classname="com.junit.exception.ExceptionThrowerTest" time="0.004">
    <system-out><![CDATA[Setting up test for exception testing...
Testing checkState with valid state...
✓ No exception thrown for valid state
Test completed.

]]></system-out>
  </testcase>
  <testcase name="testGetArrayElementValid" classname="com.junit.exception.ExceptionThrowerTest" time="0.005">
    <system-out><![CDATA[Setting up test for exception testing...
Testing getArrayElement with valid index...
✓ Valid array element retrieved: second
Test completed.

]]></system-out>
  </testcase>
  <testcase name="testThrowException" classname="com.junit.exception.ExceptionThrowerTest" time="0.007">
    <system-out><![CDATA[Setting up test for exception testing...
Testing basic exception throwing...
✓ RuntimeException thrown with correct message
Test completed.

]]></system-out>
  </testcase>
  <testcase name="testValidDivision" classname="com.junit.exception.ExceptionThrowerTest" time="0.006">
    <system-out><![CDATA[Setting up test for exception testing...
Testing valid division...
✓ Valid division completed: 10/2 = 5
Test completed.

]]></system-out>
  </testcase>
  <testcase name="testCheckStateInvalid" classname="com.junit.exception.ExceptionThrowerTest" time="0.006">
    <system-out><![CDATA[Setting up test for exception testing...
Testing checkState with invalid state...
✓ IllegalStateException thrown for invalid state
Test completed.

]]></system-out>
  </testcase>
  <testcase name="testParseNumberInvalidFormat" classname="com.junit.exception.ExceptionThrowerTest" time="0.005">
    <system-out><![CDATA[Setting up test for exception testing...
Testing parseNumber with invalid format...
✓ NumberFormatException thrown for invalid format
Test completed.

]]></system-out>
  </testcase>
</testsuite>